libs/blocks/preflight/checks/assets.js

```diff
@@ -248,50 +207,83 @@ export async function checkImageDimensions(url, area, injectVisualMetadata) {
 
   if (!assets.length) {
     return {
+      checkId: ASSETS_IDS.imageDimensions,
+      severity: ASSETS_SEVERITIES.imageDimensions,
       title: ASSETS_TITLES.AssetDimensions,
       status: STATUS.EMPTY,
       description: 'No eligible assets found (visible, non-icon, non-SVG).',
     };
   }
 
-  const assetsWithMismatch = [];
-  const assetsWithMatch = [];
-
-  if (injectVisualMetadata) area.body.classList.add('preflight-assets-analysis');
-
-  for (const asset of assets) {
+  const processedAssets = assets.map((asset) => {
     const assetData = getAssetData(asset);
-    if (injectVisualMetadata) populateAssetMeta(asset, assetData);
+    const isAssetAboveFold = isAboveFold(asset);
+    assetData.isAboveFold = isAssetAboveFold;
```

Comment:
You can do directly and then use `assetData.isAboveFold` instead of `isAssetAboveFold`
```suggestion
    assetData.isAboveFold = isAboveFold(asset);
```